src/GafferImage/ImageReader.cpp

```diff
@@ -390,6 +405,14 @@ void ImageReader::hash( const ValuePlug *output, const Context *context, IECore:
 		fileNamePlug()->hash( h );
 		h.append( OpenColorIOAlgo::currentConfigHash() );
 	}
+	else if ( output == fileValidPlug() )
+	{
+		oiioReader()->fileValidPlug()->hash( h );
+		startModePlug()->hash( h );
+		startFramePlug()->hash( h );
+		endModePlug()->hash( h );
+		endFramePlug()->hash( h );
```

Comment:
I haven't thought about it in enough detail to know if it might be equivalent in all cases, but this doesn't match the implementation of `compute()`. It's easier to verify if they mirror each other directly :

```
FrameMaskScope scope( context, this, /* clampBlack = */ true );
h = oiioReader()->fileValidPlug()->hash();
```

Passing the hash through directly like this also means that we are sharing the same cache entry for `__oiioReader.fileValid` and `ImageReader.fileValid`. This isn't really important in this case (we're just storing a bool) but can be a very worthwhile technique when the cached value is more expensive.